barrels/Semicircles/Semicircles.mc

```diff
@@ -0,0 +1,257 @@
+                                break;
+                            case :radians:
+                                result[0] = ((value[:latitude] * 0x80000000l)/Math.PI).toNumber();
+                                result[1] = ((value[:longitude] * 0x80000000l)/Math.PI).toNumber();
+                                break;
+                            default:
+                                throw new BadCoordinateFormatException("Only support formats of semicircles and radians");
+                        }
+                    } else {
+                        throw new BadCoordinateFormatException("No format specified");
+                    }
+                    break;
+                case instanceof Semicircles.Coordinate:
+                    result[0] = value._lat;
+                    result[1] = value._lon;
+                    break;
+                case instanceof Lang.Array:
+                    result[0] = value[0];
+                    result[1] = value[1];
+                    break;
+                default:
+                    // Referncing Position.Location would require the position permission.
+                    // That doesn't mean we can't check if we have a toRadians function.
+                    if (value has :toRadians) {
+                        var coords = value.toRadians();
+                        result[0] = ((coords[0] * 0x80000000l)/Math.PI).toNumber();
+                        result[1] = ((coords[1] * 0x80000000l)/Math.PI).toNumber();
+                    } else {
+                        throw new BadCoordinateFormatException("Cannot handle input type for value " + value);
+                    }
+                    break;
+            }
+            return result;
+        }
+
+        //! Computes the cosine correction for two latitudes in semicircles
+        //! @param [Toybox::Lang::Number] lat1 Latitude 1 in semicircles
+        //! @param [Toybox::Lang::Number] lat2 Latitude 2 in semicircles
+        //! @return [Toybox::Lang::Double] Cosine correction
+        private function computeCosineCorrection(lat1, lat2) {
+            var result = (lat1.toLong() + lat2.toLong()) / 2;
+            result = (result * Math.PI) / 0x80000000;
+
+            return Math.cos(result);
+        }
+
+        //! Compute a fast distance between two points
+        //! @param [Toybox::Lang::Object] pt1 Type based off the following options
+        //!    [Toybox::Position::Location] Convert the location to semicircles
+        //!    [Toybox::Lang::Array] Assumes latitude is in radians as value 0, and longitude is in radians in value 1
+        //! @param [Toybox::Lang::Object] pt2 Type based off the following options
+        //!    [Toybox::Position::Location] Convert the location to semicircles
+        //!    [Toybox::Lang::Array] Assumes latitude is in radians as value 0, and longitude is in radians in value 1
+        public function computeDistanceInMeters(pt) {
```

Comment:
The documentation for this method indicates there are two parameters but the code is only accepting one parameter. Remove the extra @param block and update the documentation to indicate a distance is computed in relation to the "this" Coordinate object.

Reply:
Done